src/components/ha-icon-picker.ts

```diff
@@ -48,17 +47,41 @@ export class HaIconPicker extends LitElement {
 
   @state() private _opened = false;
 
-  @query("ha-combo-box", true) private comboBox!: HaComboBox;
+  @state() private _filterString = "";
 
   protected render(): TemplateResult {
+    const characterCount = this._filterString.length;
+    let filteredItems: IconItem[] = [];
+    if (characterCount >= 2) {
+      const filteredItemsByKeywords: IconItem[] = [];
+
+      iconItems.forEach((item) => {
+        if (item.icon.includes(this._filterString)) {
+          filteredItems.push(item);
+          return;
+        }
+        if (item.keywords.some((t) => t.includes(this._filterString))) {
+          filteredItemsByKeywords.push(item);
+        }
+      });
+
+      filteredItems.push(...filteredItemsByKeywords);
+
+      if (!filteredItems.length) {
+        filteredItems = [{ icon: this._filterString, keywords: [] }];
+      }
+    } else {
+      filteredItems = iconItems;
+    }
```

Comment:
Can we move this to a memoized function that takes `_filterString` and `iconItems`?

Reply:
I suppose.  Would that risk over-caching if there were a lot of different filter strings attempted?

Reply:
Done.  Forgot it's only caching the last.